go.mod

```diff
@@ -1,34 +1,119 @@
 module sigs.k8s.io/cloud-provider-azure
 
-go 1.16
+go 1.17
 
 require (
 	github.com/Azure/azure-sdk-for-go v55.8.0+incompatible
 	github.com/Azure/go-autorest/autorest v0.11.22
 	github.com/Azure/go-autorest/autorest/adal v0.9.17
 	github.com/Azure/go-autorest/autorest/mocks v0.4.1
 	github.com/Azure/go-autorest/autorest/to v0.4.0
-	github.com/Azure/go-autorest/autorest/validation v0.1.0 // indirect
-	github.com/evanphx/json-patch v4.11.0+incompatible
+	github.com/evanphx/json-patch v5.6.0+incompatible
 	github.com/fsnotify/fsnotify v1.5.1
 	github.com/golang/mock v1.6.0
 	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.16.0
+	github.com/onsi/gomega v1.17.0
 	github.com/spf13/cobra v1.2.1
 	github.com/spf13/pflag v1.0.5
-	github.com/stretchr/objx v0.2.0 // indirect
 	github.com/stretchr/testify v1.7.0
-	golang.org/x/crypto v0.0.0-20210921155107-089bfa567519
-	golang.org/x/sys v0.0.0-20210630005230-0f9fa26af87c
+	golang.org/x/crypto v0.0.0-20211202192323-5770296d904e
+	golang.org/x/sys v0.0.0-20211205182925-97ca703d548d
 	k8s.io/api v0.22.4
 	k8s.io/apimachinery v0.22.4
 	k8s.io/apiserver v0.22.4
 	k8s.io/client-go v0.22.4
 	k8s.io/cloud-provider v0.22.4
 	k8s.io/component-base v0.22.4
 	k8s.io/controller-manager v0.22.4
-	k8s.io/klog/v2 v2.10.0
+	k8s.io/klog/v2 v2.9.0
```

Comment:
Why downgrade this?

Reply:
I encounter a build error which indicates that custom logger is not compatible. So I checked the go.mod file of k8s.io/apiserver, changed the version to 2.9.0 and it worked. But I can't reproduce it now. I will revert it in new commit.